examples/dek-reuse/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import {SDK, DeviceDetails, initialize} from "@ironcorelabs/ironnode"
+import * as crypto from "crypto"
+import * as fs from "fs"
+
+
+const AES_ALGORITHM = "aes-256-gcm"
+const AES_SYMMETRIC_KEY_LENGTH = 32
+const AES_IV_LENGTH = 12
+const AES_GCM_TAG_LENGTH = 16
+    
+const DATA_BLOCK_LENGTH = 256
+const NUM_DATA_BLOCKS = 3
+    
+
+async function writeDataFile(dek: Buffer, fileNum: number) {
+    const dataBlock = crypto.randomBytes(DATA_BLOCK_LENGTH).toString("base64")
+    const iv = crypto.randomBytes(AES_IV_LENGTH)
+    const cipher = crypto.createCipheriv(AES_ALGORITHM, dek, iv)
+    const encryptedData =
+          Buffer.concat([iv, cipher.update(dataBlock), cipher.final(), cipher.getAuthTag()])
+    const dataBlockFname = `./dataFile${fileNum}.txt`
+    fs.writeFileSync(dataBlockFname, dataBlock)
+    const encDataFname = `./dataFile${fileNum}.enc`
+    fs.writeFileSync(encDataFname, encryptedData)
+}
+
+
+async function writeInitialData(ironnode: SDK) {
+    // Generate a random AES key to use to encrypt multiple blocks of data. We refer to this as
+    // the "Data Encryption Key", or DEK.
+    const dek = crypto.randomBytes(AES_SYMMETRIC_KEY_LENGTH)
+    
+    // Now generate some blocks of data to encrypt. Each one is just random bytes which are
+    // base64 encoded so it's easier to look at them. For each one, generate a random IV
+    // (initialization vector), then encrypt the block using AES256-GCM with the provided key
+    // and the IV. Write each input data block to its own file, then write the encrypted block
+    // to a corresponding file.
+    for (let i = 0; i < NUM_DATA_BLOCKS; i++) {
+        await writeDataFile(dek, i)
+    }
+    
+    // Now encrypt the AES key using ironnode. We don't pass in any document options - it defaults
+    // to an empty document name, random document ID, and encrypting only to the current user -
+    // you could encrypt to additional users or groups if desired. Persist this along with the
+    // encrypted data files.
```

Comment:
"Persist this EDEK along with the data files encrypted using its contained DEK.~